dbfit-java/snowflake/src/main/java/dbfit/environment/SnowflakeEnvironment.java

```diff
@@ -0,0 +1,207 @@
+            return columns;
+        }
+    }
+
+    // List interface has sequential search, so using list instead of array to
+    // map types
+    private static List<String> stringTypes = Arrays.asList("VARCHAR", "CHAR", "TEXT", "STRING");
+    private static List<String> booleanTypes = Arrays.asList("BOOLEAN");
+
+    private static List<String> decimalTypes = Arrays.asList("DECIMAL", "NUMBER", "NUMERIC",
+            "INT", "INTEGER", "TINYINT", "SMALLINT" , "BIGINT");
+    private static List<String> timestampTypes = Arrays.asList("DATETIME", "TIMESTAMP_LTZ", "TIMESTAMP_NTZ",
+            "TIMESTAMP_TZ" );
+    private static List<String> dateTypes = Arrays.asList("DATE");
+    private static List<String> timeTypes = Arrays.asList("TIME");
+
+     private static List<String> variantTypes = Arrays.asList("VARIANT");
+
+    private static int getSqlType(String dataType) {
+        dataType = normaliseTypeName(dataType);
+
+        if (stringTypes.contains(dataType))
+            return java.sql.Types.VARCHAR;
+        if (decimalTypes.contains(dataType))
+            return Types.DECIMAL;
+        if (timestampTypes.contains(dataType))
+            return java.sql.Types.TIMESTAMP;
+        if (dateTypes.contains(dataType))
+            return java.sql.Types.DATE;
+        if (timeTypes.contains(dataType))
+            return java.sql.Types.TIME;
+        if (booleanTypes.contains(dataType))
+            return java.sql.Types.BOOLEAN;
+        if (variantTypes.contains(dataType))
+            return java.sql.Types.BOOLEAN;
+
+        throw new UnsupportedOperationException("Type " + dataType
+                + " is not supported");
+    }
+
+    public Class<?> getJavaClass(String dataType) {
+        dataType = normaliseTypeName(dataType);
+        if (stringTypes.contains(dataType))
+            return String.class;
+        if (decimalTypes.contains(dataType))
+            return BigDecimal.class;
+        if (timestampTypes.contains(dataType))
+            return java.sql.Timestamp.class;
+        if (dateTypes.contains(dataType))
+            return java.sql.Date.class;
+        if (timeTypes.contains(dataType))
+            return java.sql.Time.class;
+        if (booleanTypes.contains(dataType))
+            return Boolean.class;
+        if (variantTypes.contains(dataType))
+            return String.class;
+
+        throw new UnsupportedOperationException("Type " + dataType
+                + " is not supported");
+    }
+
+    private static String normaliseTypeName(String dataType) {
+        return  dataType.toUpperCase().trim();
+    }
+
+    public Map<String, DbParameterAccessor> getAllProcedureParameters(
+            String procName) throws SQLException {
+
+        String[] qualifiers = NameNormaliser.normaliseName(procName).split(
+                "\\.");
+        String qry = " select FUNCTION_LANGUAGE, ARGUMENT_SIGNATURE, DATA_TYPE from information_schema.functions where lower(function_name)=?";
```

Comment:
Very long line - I would rather suggest splitting it. Do we need the blank space before `select` here?